pxe-server/build/Dockerfile

```diff
@@ -8,10 +8,75 @@ RUN apt-get update && \
         dnsmasq=2.90-4~deb12u1 && \
     rm -rf /var/lib/apt/lists/*
 
+# Download sources for dependent packages and create tar archive
+RUN mkdir -p /sources && \
+    # Install dpkg-dev for source package handling
+    apt-get update && \
+    apt-get install -y --no-install-recommends dpkg-dev tar && \
+    # Create source repositories
+    echo "deb-src http://deb.debian.org/debian bookworm main" > /etc/apt/sources.list.d/debian-sources.list && \
+    echo "deb-src http://deb.debian.org/debian-security bookworm-security main" >> /etc/apt/sources.list.d/debian-sources.list && \
+    # Update with --allow-insecure-repositories to avoid signing conflicts
```

Comment:
should we handle apt failures 

```
apt-get source "$package=$version" 2>/dev/null || \
apt-get source "$package" 2>/dev/null || \
echo "No source available for $package";
```

Reply:
|| at the end ensures continuation even if it fails.